bitcoin/src/bip32.rs

```diff
@@ -1007,29 +1385,41 @@ mod tests {
     ) {
         let mut sk = Xpriv::new_master(network, seed).unwrap();
         let mut pk = Xpub::from_priv(secp, &sk);
-
         // Check derivation convenience method for Xpriv
         assert_eq!(&sk.derive_priv(secp, &path).to_string()[..], expected_sk);
 
+        // Tries to convert the path into a normal path, if it fails, it should return an error
+        match path.to_normal() {
+            Ok(path) => {
+                assert_eq!(&pk.derive_pub(secp, &path).to_string()[..], expected_pk);
+            }
+            Err(e) => {
+                assert_eq!(e, Error::CannotDeriveFromHardenedKey);
+            }
+        }
```

Comment:
Whether the test should fail should be carried in `expected_pk` variable (e.g. as `Option`).

Reply:
I did a adaptation for the new methods, the old one does the same thing... 
I do not understand this piece of the `test_path()` too.
But since https://github.com/rust-bitcoin/rust-bitcoin/discussions/3057 exist, we could consider some refactor on it ?

Reply:
> the old one does the same thing...

Oh, that sucks. Could be a different PR.